pandas/core/frame.py

```diff
@@ -4527,9 +4527,12 @@ def _sanitize_column(self, value) -> ArrayLike:
         """
         self._ensure_valid_index(value)
 
-        # We should never get here with DataFrame value
-        if isinstance(value, Series):
+        # We can get there through isetitem with a DataFrame
+        # or through loc single_block_path
+        if isinstance(value, DataFrame):
             return _reindex_for_setitem(value, self.index)
+        elif is_dict_like(value):
+            return _reindex_for_setitem(Series(value), self.index)
```

Comment:
this includes the changes from #47361, so should backport that first (I think) so that we include test_setitem_aligning_dict_with_index in the backport also

Reply:
and move that release note too (and update the milestones etc)

Reply:
but first need to confirm if that's also a regression (I can do that tomorrow)

Reply:
This was not a regression unfortunately, did not work on 1.3.5 either